static/js/publisher/release/components/modal.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { PropTypes } from "prop-types";
+
+import { closeModal } from "../actions/modal";
+
+import * as actions from "../actions/";
+
+class ModalActionButton extends Component {
+  constructor(props) {
+    super(props);
+
+    this.onClickHandler = this.onClickHandler.bind(this);
+    this.state = {
+      loading: false
+    };
+  }
+
+  onClickHandler() {
+    const { onClickAction, dispatch } = this.props;
+
+    if (onClickAction.reduxAction) {
+      const { reduxAction } = onClickAction;
+      if (
+        this.props[reduxAction] &&
+        typeof this.props[reduxAction] == "function"
+      ) {
+        // If an action is passed, perform the specific action
+        this.props[onClickAction.reduxAction]();
+      }
+    } else {
+      // Otherwise dispatch the action object
+      dispatch(onClickAction);
+    }
+
+    this.setState({
+      loading: true
```

Comment:
The loading is specifically for the button (not the entire modal) so we'd probably want to add another action & reducer.
We'd have to call it in a similar way, unless we added a dispatch to every action that *could* trigger the state change. But then we might get some odd behaviour like the "Submit" button turning into a "Loading..." button because some background fetch/ state change triggered the modal button state change.

Reply:
OK. Right. Button would need to know what is loading.
Let's not complicate it and leave as it is.